File: CodeList/CodeList19/MainActivity.java
package com.example.myviewmodelapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;


public class MainActivity extends AppCompatActivity {

    Button btn;
    TextView txt;
    String myName = "";
    MyViewModel myModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        txt = (TextView)findViewById(R.id.txt);
        btn = (Button)findViewById(R.id.btn);
        myModel = new ViewModelProvider(this).get(MyViewModel.class);
        txt.setText(myModel.myName);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                change_name("Ngoc Minh");
            }
        });
    }
    public void change_name(String newName){
        myModel.myName = newName;
        txt.setText(myModel.myName);
    }

}

File: Part2/MainActivity.kt
package com.example.admin.myfirstkotlinapp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        numList.text = numList(5)
        evenList.text = evenList(5)
        oddList.text = oddList(5)
    }
     fun numList (n:Int):String
     {
         var S = 0
         var str = ""
         for(i in 1..n)
         {
             S = S + i
             str += " $i"
         }

         return "Sum of [ $str ] = $S"
     }
    fun evenList (n:Int):String
    {
        var S = 0
        var str = ""
        for(i in 1..n)
        {
            if(i%2 == 0)
            {
                S = S + i
                str += " $i"
            }
        }

        return "Sum of [ $str ] = $S"
    }
    fun oddList (n:Int):String
    {
        var S = 0
        var str = ""
        for(i in 1..n)
        {
            if(i%2 != 0)
            {
                S = S + i
                str += " $i"
            }
        }

        return "Sum of [ $str ] = $S"
    }
}
